homeassistant/components/version/config_flow.py

```diff
@@ -0,0 +1,202 @@
+        self._entry_data.update(user_input)
+
+        if not self.show_advanced_options or user_input[CONF_SOURCE] in (
+            HaVersionSource.LOCAL,
+            HaVersionSource.HAIO,
+        ):
+            return self.async_create_entry(
+                title=self._config_entry_name,
+                data=self._entry_data,
+            )
+
+        return await self.async_step_version_source()
+
+    async def async_step_version_source(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Handle the version_source step."""
+        if user_input is None:
+            if self._entry_data[CONF_SOURCE] in (
+                HaVersionSource.SUPERVISOR,
+                HaVersionSource.CONTAINER,
+            ):
+                data_schema = vol.Schema(
+                    {
+                        vol.Required(
+                            CONF_CHANNEL, default=DEFAULT_CHANNEL.title()
+                        ): vol.In(VALID_CHANNELS),
+                    }
+                )
+                if self._entry_data[CONF_SOURCE] == HaVersionSource.SUPERVISOR:
+                    data_schema = data_schema.extend(
+                        {
+                            vol.Required(CONF_IMAGE, default=DEFAULT_IMAGE): vol.In(
+                                VALID_IMAGES
+                            ),
+                            vol.Required(CONF_BOARD, default=DEFAULT_BOARD): vol.In(
+                                VALID_BOARDS
+                            ),
+                        }
+                    )
+                else:
+                    data_schema = data_schema.extend(
+                        {
+                            vol.Required(CONF_IMAGE, default=DEFAULT_IMAGE): vol.In(
+                                VALID_CONTAINER_IMAGES
+                            )
+                        }
+                    )
+            else:
+                data_schema = vol.Schema({vol.Required(CONF_BETA, default=False): bool})
+
+            return self.async_show_form(
+                step_id=STEP_VERSION_SOURCE,
+                data_schema=data_schema,
+                description_placeholders={
+                    ATTR_VERSION_SOURCE: self._entry_data[CONF_VERSION_SOURCE]
+                },
+            )
+        self._entry_data.update(user_input)
+        self._entry_data[CONF_CHANNEL] = self._entry_data[CONF_CHANNEL].lower()
+
+        return self.async_create_entry(
+            title=self._config_entry_name, data=self._entry_data
+        )
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        self._entry_data = _convert_imported_configuration(import_config)
+
+        for entry in self._async_current_entries():
+            if _fingerprint(entry.data) == _fingerprint(self._entry_data):
```

Comment:
Would this helper give the same result?

https://github.com/home-assistant/core/blob/c5d8792c3487e9b418b1e7d623bf59e7dbddd6b7/homeassistant/config_entries.py#L1201